js-foundations.js

```diff
@@ -1,50 +1,151 @@
+const divisibleBy = (num) => {
+    if(num%3 === 0){
+        return `${num} is divisble by 3`
+    } else {
+        return `${num} is not divisibleBy`
+    }
+}
+console.log(divisibleBy(15))
+console.log(divisibleBy(0))
+console.log(divisibleBy(385))
+console.log(divisibleBy(-7))
+console.log(divisibleBy(-9))
+
+// // 2. Create an object called helloMe. Include your firstName, lastName and TWO OTHER properties of you. Add a method to your object that will return a sentence using the object properties to describe you.
+const helloMe = {
+    firstName: 'Christi', 
+    lastName: 'Wright', 
+    origin: 'Bay Area', 
+    alum: 'UCSD',
+    getData: function () {return `${helloMe.firstName} ${helloMe.lastName} is from the ${helloMe.origin} and went to college at ${helloMe.alum}.`}
+}
+
+console.log(helloMe.getData())
 
 // ------------------------------------- Consider the variable:
 
 var randomNouns = ["streetlamp", "potato", "teeth", "conclusion", "nephew", "temperature", "database"]
 
-// 3a. Write a function that returns every other item from the array as one string. Expected output: "streetlamp teeth nephew database"
+// // 3a. Write a function that returns every other item from the array as one string. Expected output: "streetlamp teeth nephew database"
+ 
+//filter - need to update to function, currently returning the whole array
+// const arrayToString = (array) => {
+//     return array.filter(word => word ).join(" ") 
+//  }
+// console.log(arrayToString(randomNouns))
+
+const arrayToString2 = (array) => {
+    let newArray = []
+    for (let i = 0; i<array.length; i++){
+        if (i%2 === 0){
+        newArray.push(array[i])
+        }    
+    }
+    return newArray.join(" ")
+}
+console.log(arrayToString2(randomNouns))
 
+// // 3b. Create a function that takes in the variable and returns the array with all the words capitalized. Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew", "Temperature", "Database"]
+// //why is my array printing each element on a new line?
+const capitalizer = (array) => {
+    newArray = []
+    for(let i = 0; i<array.length; i++){
+        newArray.push(array[i].charAt(0).toUpperCase() + array[i].slice(1))
+    }
+    return newArray
+}
 
+console.log(capitalizer(randomNouns))
 
-// 3b. Create a function that takes in the variable and returns the array with all the words capitalized. Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew", "Temperature", "Database"]
+// // option... returns undefined currently....why? this iteration should have cleared up comp's call confusion..
+// function capitalizer2(array) {
+//     return array.forEach(function(element) {element.charAt(0).toUpperCase() + element.slice(1)})
+// }
 
+// console.log(capitalizer2(randomNouns))
 
+// const capitalizer2 = (array) => {
+//     return array.forEach(element => {element.replace(element[0], element[0].toUpperCase()})
+// }
 
```

Comment:
Basically forEach is being a pain. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach
If you replace forEach with map and return your nested function, your code works just fine.